lib/fluent/plugin/in_wmi.rb

```diff
@@ -13,14 +14,29 @@ def configure(conf)
 
     def start
       super
-      tag = "wmi.test"
+      wmi = WmiLite::Wmi.new
+      instances = wmi.instances_of('Win32_Processor')
+      tag = "monitor.wmi"
       time = Fluent::Engine.now
-      record = {"plugin"=>"wmi"}
+      record = instance_to_hash(instances)
       router.emit(tag, time, record)
     end
 
     def shutdown
       super
     end
+
+  private
+    def instance_to_hash(instances)
+      a = []
+      instances.each do |instance|
+        a << instance.property_map
+      end
+      return a
+    end
   end
 end
+
+class WmiLite::Wmi::Instance
+  attr_reader :property_map

```

Comment:
I'm kinda curious about this attribute overriding.

Since this seems to be a very common usage of the WMI library,
I am wondering if it's worth reporting to the upstream project...

Reply:
Yes, I think so and I will make PR to upstream about this idea.
But this hack is not smart well, maybe rewrite by incuding Enumerable is better way.